modules/account-lib/src/coin/stx/utils.ts

```diff
@@ -172,3 +185,56 @@ export function isValidPrivateKey(prv: string): boolean {
 function allHexChars(maybe: string): boolean {
   return maybe.match(/^[0-9a-f]+$/i) !== null;
 }
+
+/**
+ * Checks if raw transaction can be deserialized
+ *
+ * @param {unknown} rawTransaction - transaction in raw hex format
+ * @returns {boolean} - the validation result
+ */
+export function isValidRawTransaction(rawTransaction: unknown): boolean {
+  try {
+    if (typeof rawTransaction === 'string') {
+      deserializeTransaction(BufferReader.fromBuffer(Buffer.from(removeHexPrefix(rawTransaction), 'hex')));
+    } else {
+      return false;
+    }
+  } catch (e) {
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Checks for valid contract address
+ *
+ * @param {string} addr - contract deployer address
+ * @param {StacksNetwork} network - network object
+ * @returns {boolean} - the validation result
+ */
+export function isValidContractAddress(addr: string, network: StacksNetwork): boolean {
+  if (network.isMainnet()) {
```

Comment:
Thinking this through we might want to `else if network.isTestnet())` for the else case. Otherwise we won't be able to have the contract call testing in develop.

If network is develop (local), I think we can just say all contract addresses are valid. Just because I dont think we'll know the staking contract address until we deploy it. @marktoda do you see any issues with this?

Reply:
That sounds reasonable